org.telegram.desktop.json

```diff
@@ -0,0 +1,276 @@
+{
+    "id": "org.telegram.desktop",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.9",
+    "sdk": "org.kde.Sdk",
+    "command": "telegram-desktop",
+    "rename-desktop-file": "telegramdesktop.desktop",
+    "rename-appdata-file": "telegramdesktop.appdata.xml",
+    "rename-icon": "telegram-desktop",
+    "finish-args": [ "--share=ipc",
+                     "--share=network",
+                     "--socket=x11",
+                     "--socket=pulseaudio",
+                     "--talk-name=org.kde.StatusNotifierWatcher",
```

Comment:
I believe you need to set this for indicator to work: `"--env=XDG_CURRENT_DESKTOP=Unity",`

https://github.com/telegramdesktop/tdesktop/blob/c1aa1c5a0f26b29fce1307ba455f9887e7c1c1ec/Telegram/SourceFiles/platform/linux/linux_desktop_environment.cpp#L123

Reply:
We have telegram compiled with disabled unity integration, I don't remember the reason why is that. Still, if you  set XDG_CURRENT_DESKTOP to Unity, then none of Qt platform themes would work.

Reply:
> Still, if you set XDG_CURRENT_DESKTOP to Unity, then none of Qt platform themes would work.

That is odd. Maybe just patch it to always try AppIndicator. The other Unity integration was things like badges that we don't need for now.

Reply:
That's not odd, Qt picks platform themes based on running desktop. E.g. if you are running Gnome, then QGnomePlatform is used. If you set XDG_CURRENT_DESKTOP to Unity, then Qt wouldn't use it at all. Telegram also itself does some detection when to run appindicator and when not, I don't want to end up patching it, if it's broken then it's better to report it and fix in upstream.

Reply:
I believe Telegram should just always use AppIndicator when available, do you want to report that?

Reply:
That's what is telegram trying to do currently and I don't want to report that, because I don't see a problem at all.

https://github.com/telegramdesktop/tdesktop/blob/c1aa1c5a0f26b29fce1307ba455f9887e7c1c1ec/Telegram/SourceFiles/platform/linux/main_window_linux.cpp#L385

Reply:
`	tryAppIndicator = !DesktopEnvironment::PreferAppIndicatorTrayIcon();`

They clearly check if they should or should not based upon desktop. Which is silly and wrong considering AppIndicators can be used by Plasma, GNOME, etc.

Reply:
That just says where AppIndicator should be preffered over QSystemTray. I think the problem here is that there should be a better detection where to use QSystemTray icon, because obviously it won't work on Gnome because you removed support for it. Forcing to try AppIndicator on all desktops is also a wrong solution, we don't want to use it in Plasma as we can have native support with QSystemTray.

Reply:
Alright, this isn't a blocker anyway just something that I believe could use improvements.